utils/comment_test.go

```diff
@@ -43,24 +43,177 @@ func TestGetFrogbotReviewComments(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			output := getFrogbotComments(writer, tc.existingComments)
+			output := getFrogbotComments(tc.existingComments)
 			assert.ElementsMatch(t, tc.expectedOutput, output)
 		})
 	}
 }
 
+func TestGroupSimilarJasIssues(t *testing.T) {
+	testCases := []struct {
+		name          string
+		issues        []formats.SourceCodeRow
+		groupedIssues []similarIssues
+	}{
+		{
+			name: "No issues",
+		},
+		{
+			name: "Single issue",
+			issues: []formats.SourceCodeRow{
+				{
+					SeverityDetails: formats.SeverityDetails{Severity: "High"},
+					Location:        formats.Location{File: "file1", StartLine: 1, StartColumn: 10, EndLine: 2, EndColumn: 11, Snippet: "snippet"},
+					Finding:         "finding1",
+					ScannerInfo:     formats.ScannerInfo{RuleId: "rule1"},
+				},
+			},
+			groupedIssues: []similarIssues{
+				{
+					formats.Location{File: "file1", StartLine: 1, StartColumn: 10, EndLine: 2, EndColumn: 11, Snippet: "snippet"},
+					[]formats.SourceCodeRow{
+						{
+							SeverityDetails: formats.SeverityDetails{Severity: "High"},
+							Location:        formats.Location{File: "file1", StartLine: 1, StartColumn: 10, EndLine: 2, EndColumn: 11, Snippet: "snippet"},
+							Finding:         "finding1",
+							ScannerInfo:     formats.ScannerInfo{RuleId: "rule1"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Multiple issues - no similar issues",
```

Comment:
how do you define "similar issues"? similar in what?
in the func description it seems like you define similar issues as 'same location, same id but different rule'.
from this test case you have 2 issues with everything similar BUT the ruldId and you considered it as "no similar issues".
Please define in the groupSimilarJasIssues description what is exactly "similar issues" since the test cases names are not making sense to me